ClaimOwnership/handler.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  IResponseErrorInternal,
+  IResponseSuccessJson,
+  ResponseErrorInternal,
+  ResponseSuccessJson
+} from "@pagopa/ts-commons/lib/responses";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "@pagopa/ts-commons/lib/request_middleware";
+import * as express from "express";
+import { flow, pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import { ContextMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredParamMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_param";
+import {
+  NonEmptyString,
+  OrganizationFiscalCode
+} from "@pagopa/ts-commons/lib/strings";
+import { Pool, QueryResult, QueryResultRow } from "pg";
+import { Context } from "@azure/functions";
+import { knex } from "knex";
+import { IConfig, IDecodableConfigPostgreSQL } from "../utils/config";
+import {
+  IDbError,
+  toPostgreSQLError,
+  toPostgreSQLErrorMessage
+} from "../models/DomainErrors";
+import { SubscriptionStatus } from "../GetOwnershipClaimStatus/handler";
+import { queryDataTable } from "../utils/db";
+import { OrganizationQueueItem } from "./types";
+
+type Handler = (
+  context: Context,
+  organizationFiscalCode: OrganizationFiscalCode,
+  delegateId: NonEmptyString
+) => Promise<IResponseSuccessJson<void> | IResponseErrorInternal>;
+
+export const updateSqlSubscription = (dbConfig: IDecodableConfigPostgreSQL) => (
+  organizationFiscalCode: OrganizationFiscalCode,
+  sourceId: NonEmptyString
+): NonEmptyString =>
+  knex({
+    client: "pg"
+  })
+    .withSchema(dbConfig.DB_SCHEMA)
+    .table(dbConfig.DB_TABLE)
+    .where({ organizationFiscalCode })
+    .where({ sourceId })
+    .where("status", "!=", SubscriptionStatus.COMPLETED)
+    .update({
+      status: SubscriptionStatus.PROCESSING
+    })
+    .from(dbConfig.DB_TABLE)
+    .toQuery() as NonEmptyString;
+
+export const updateSubscriptionsByOrganizationFiscalCodeAndSourceId = (
+  config: IConfig,
+  connect: Pool
+) => (
+  organizationFiscalCode: OrganizationFiscalCode,
+  sourceId: NonEmptyString
+): TE.TaskEither<IDbError, QueryResult<QueryResultRow>> =>
+  pipe(
+    updateSqlSubscription(config)(organizationFiscalCode, sourceId),
+    sql => queryDataTable(connect, sql),
+    TE.mapLeft(flow(toPostgreSQLErrorMessage, toPostgreSQLError))
+  );
+
+export const organizationMessageToQueue = (context: Context) => (
```

Comment:
Naming: what's this function for? Is it clear that it has a side effect (it writes to the queue)? It makes me think it is just a conversion from a structure X to Y.

Please decide whether:
* to make the function pure, thus moving the side effect outside
* to keep the function to encapsulate side effects, and then it needs to have an explicit name

Reply:
The function now is splitted in 2 easier function, one to create the message and one to write the message to the Queue. Maybe it's better in this way: d839eb9 ?